processors/filterprocessor/filter_processor_test.go

```diff
@@ -15,6 +15,141 @@ import (
+    - tag: txn.arcv
+      expression-type: regex
+      expression: "` + sampleAddr2.String() + `"
+`
+
+	fpBuilder, err := processors.ProcessorBuilderByName(implementationName)
+	assert.NoError(t, err)
+
+	fp := fpBuilder.New()
+	err = fp.Init(context.Background(), &conduit.PipelineInitProvider{}, plugins.PluginConfig(sampleCfgStr), logrus.New())
+	assert.NoError(t, err)
+
+	bd := data.BlockData{}
+	bd.Payset = append(bd.Payset,
+
+		transactions.SignedTxnInBlock{
+			SignedTxnWithAD: transactions.SignedTxnWithAD{
+				SignedTxn: transactions.SignedTxn{
+					AuthAddr: sampleAddr1,
+				},
+			},
+		},
+		transactions.SignedTxnInBlock{
+			SignedTxnWithAD: transactions.SignedTxnWithAD{
+				SignedTxn: transactions.SignedTxn{
+					AuthAddr: sampleAddr1,
+					Txn: transactions.Transaction{
+						PaymentTxnFields: transactions.PaymentTxnFields{
+							Receiver: sampleAddr2,
+						},
+						Header: transactions.Header{
+							Sender: sampleAddr2,
+						},
+						AssetTransferTxnFields: transactions.AssetTransferTxnFields{
+							AssetCloseTo: sampleAddr2,
+						},
+					},
+				},
+			},
+		},
+		transactions.SignedTxnInBlock{
+			SignedTxnWithAD: transactions.SignedTxnWithAD{
+				SignedTxn: transactions.SignedTxn{
+					AuthAddr: sampleAddr1,
+					Txn: transactions.Transaction{
+						AssetTransferTxnFields: transactions.AssetTransferTxnFields{
+							AssetSender: sampleAddr3,
+						},
+						PaymentTxnFields: transactions.PaymentTxnFields{
+							Receiver: sampleAddr3,
+						},
+					},
+				},
+			},
+		},
+		transactions.SignedTxnInBlock{
+			SignedTxnWithAD: transactions.SignedTxnWithAD{
+				SignedTxn: transactions.SignedTxn{
+					AuthAddr: sampleAddr1,
+					Txn: transactions.Transaction{
+						PaymentTxnFields: transactions.PaymentTxnFields{
+							Receiver: sampleAddr2,
+						},
+						Header: transactions.Header{
+							Sender: sampleAddr2,
+						},
+						AssetTransferTxnFields: transactions.AssetTransferTxnFields{
+							AssetSender:   sampleAddr3,
+							AssetCloseTo:  sampleAddr2,
+							AssetReceiver: sampleAddr2,
+						},
+					},
+				},
+			},
+		},
+		// The one transaction that will be allowed through
+		transactions.SignedTxnInBlock{
+			SignedTxnWithAD: transactions.SignedTxnWithAD{
+				SignedTxn: transactions.SignedTxn{
+					AuthAddr: sampleAddr2,
+					Txn: transactions.Transaction{
+						PaymentTxnFields: transactions.PaymentTxnFields{
+							Receiver: sampleAddr2,
+						},
+						Header: transactions.Header{
+							Sender: sampleAddr2,
+						},
+						AssetTransferTxnFields: transactions.AssetTransferTxnFields{
+							AssetSender:   sampleAddr1,
+							AssetCloseTo:  sampleAddr2,
+							AssetReceiver: sampleAddr2,
+						},
+					},
+				},
+			},
+		},
+	)
+
+	output, err := fp.Process(bd)
+	assert.NoError(t, err)
+	assert.Equal(t, len(output.Payset), 1)
+	assert.Equal(t, output.Payset[0].SignedTxnWithAD.SignedTxn.Txn.PaymentTxnFields.Receiver, sampleAddr2)
```

Comment:
nit: maybe assign the txn to a var and then assert equal on output.Payset[0] and that txn.